iphone/Maps/UI/Search/Filters/MWMSearchHotelsFilterViewController.mm

```diff
@@ -83,67 +114,180 @@ + (MWMSearchHotelsFilterViewController *)controller
 - (void)viewDidLoad
 {
   [super viewDidLoad];
-  configButton(self.ratingAny, L(@"booking_filters_rating_any"), nil);
-  configButton(self.rating7, L(@"7.0+"), L(@"booking_filters_ragting_good"));
-  configButton(self.rating8, L(@"8.0+"), L(@"booking_filters_rating_very_good"));
-  configButton(self.rating9, L(@"9.0+"), L(@"booking_filters_rating_excellent"));
+  self.tableView.estimatedRowHeight = 48;
+  self.tableView.rowHeight = UITableViewAutomaticDimension;
+}
 
-  configButton(self.price1, L(@"$"), nil);
-  configButton(self.price2, L(@"$$"), nil);
-  configButton(self.price3, L(@"$$$"), nil);
+- (void)initialRatingConfig
+{
+  MWMFilterRatingCell * rating = self.rating;
+  configButton(rating.any, L(@"booking_filters_rating_any"), nil);
+  configButton(rating.good, L(@"7.0+"), L(@"booking_filters_ragting_good"));
+  configButton(rating.veryGood, L(@"8.0+"), L(@"booking_filters_rating_very_good"));
+  configButton(rating.excellent, L(@"9.0+"), L(@"booking_filters_rating_excellent"));
+  rating.any.selected = YES;
+  [self resetRating];
+}
+
+- (void)initialPricaCategoryConfig
+{
+  MWMFilterPriceCategoryCell * price = self.price;
+  configButton(price.one, L(@"$"), nil);
+  configButton(price.two, L(@"$$"), nil);
+  configButton(price.three, L(@"$$$"), nil);
+  price.one.selected = NO;
+  price.two.selected = NO;
+  price.three.selected = NO;
+  [self resetPriceCategory];
+}
+
+- (void)reset
+{
+  [self resetRating];
+  [self resetPriceCategory];
+  [self resetTypes];
+}
+
+- (void)resetRating
+{
+  MWMFilterRatingCell * rating = self.rating;  
+  rating.any.selected = YES;
+  rating.good.selected = NO;
+  rating.veryGood.selected = NO;
+  rating.excellent.selected = NO;
+}
 
-  [self changeRating:self.ratingAny];
+- (void)resetPriceCategory
+{
+  MWMFilterPriceCategoryCell * price = self.price;
+  price.one.selected = NO;
+  price.two.selected = NO;
+  price.three.selected = NO;
+}
 
-  self.price1.selected = NO;
-  self.price2.selected = NO;
-  self.price3.selected = NO;
+- (void)resetTypes
+{
+  m_selectedTypes.clear();
+  [self.type.collectionView reloadData];
 }
 
 - (shared_ptr<search::hotels_filter::Rule>)rules
 {
   using namespace search::hotels_filter;
+  MWMFilterRatingCell * rating = self.rating;
   shared_ptr<Rule> ratingRule;
-  if (self.rating7.selected)
+  if (rating.good.selected)
     ratingRule = Ge<Rating>(7.0);
-  else if (self.rating8.selected)
+  else if (rating.veryGood.selected)
```

Comment:
I would not do that. Those strings are passed to localization function as a key.
We can some sort of "string from double" builder for these keys, however as to me it will reduce readability.